### PR TITLE
try to fix pdb rendering on rhs

### PR DESCRIPTION
**Pull request by reskyner**

### CODE REVIEW

File: fragalysis/settings.py [hunk 1]
```diff
@@ -239,4 +239,5 @@
 EMAIL_USE_TLS = True
 EMAIL_PORT = 587
 EMAIL_HOST_USER = os.environ.get("EMAIL_USER")
-EMAIL_HOST_PASSWORD = os.environ.get("EMAIL_PASSWORD")
\ No newline at end of file
+EMAIL_HOST_PASSWORD = os.environ.get("EMAIL_PASSWORD")
+DEBUG=True
```
Comment (reskyner) [inline fragalysis/settings.py:243]: Remove this

File: media_serve/urls.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 from . import views
 
 urlpatterns = [url(r"^pdbs/(?P<file_path>.+)", views.prot_download, name="get_protein"),
+               #url(r"^uploaded_pdbs/(?P<file_path>.+)", views.uploaded_prot_download, name="get_uploaded_protein"),
```
Comment (reskyner) [inline media_serve/urls.py:6]: delete if not needed

File: viewer/compound_set_upload.py [hunk 3]
```diff
@@ -51,22 +52,52 @@ def dataType(str):
...
-def get_prot(mol, compound_set, zfile):
-    pdb_option = mol.GetProp('ref_pdb')
-    name = pdb_option.split('/')[-1]
+def process_pdb(pdb_code, target, zfile):
+    pdb_fn  = zfile[pdb_code].split('/')[-1]
+    pdb_fp =  zfile[pdb_code]
+    
+    # Move and save the protein pdb from tmp to pdbs folder
+    # pdb may have already been moved and Protein object created
+    try:
```
Comment (reskyner) [inline viewer/compound_set_upload.py:61]: Is there a way to do it other than with a try/except? e.g. 

```python
prot_objs = Protein.objects.filter(code=pdb_code)
if len(prot_obj)==0:
    # do the new bit here
elif len(prot_objs)==1:
    prot_obj = prot_objs[0]
else:
    raise Exception('something went wrong - multiple matching proteins found')

```

File: viewer/compound_set_upload.py [hunk 4]
```diff
@@ -126,12 +157,14 @@ def set_mol(mol, compound_set, filename, zfile=None):
...
-        # Save the compound set
-        filename = settings.MEDIA_ROOT + prot_field
-        prot_field = filename
-        compound_set.save()
+    prot_obj = get_prot(mol, target, compound_set, zfile)
+    # if 'tmp' in prot_field:
+    #     # Save the compound set
+    #     filename = settings.MEDIA_ROOT + prot_field
+    #     prot_field = filename
+    #     compound_set.save()
```
Comment (reskyner) [inline viewer/compound_set_upload.py:165]: delete if not needed

File: viewer/models.py [hunk 5]
```diff
@@ -275,10 +275,19 @@ class ComputedMolecule(models.Model):
     # comes from compound
     #original_smiles = models.CharField(max_length=255)
     # link to pdb file for prot structure
+    # instead link to a Protein
+    protein = models.ForeignKey(Protein, null=True)
```
Comment (reskyner) [inline viewer/models.py:279]: add `on_delete=models.CASCADE` - this is what is making the build fail on the PR

File: viewer/views.py [hunk 6]
```diff
@@ -235,10 +236,34 @@ def post(self, request):
...
+                                                        ContentFile(zf.read(filename)))
+                            zfile[code] = path
+                          
+                            # # create Protein object, code=filename minus .pdb, pdb_info = file (default_storage.save)
+                            # prot = Protein()
+                            # prot.code = code
+                            # target_obj = Target.objects.get(title=target)
+                            # prot.target_id = target_obj
+                            # prot.pdb_info = path
+                            # prot.save()
```
Comment (reskyner) [inline viewer/views.py:266]: delete comments if not needed

**Comment by Waztom** [comment]
Will close this PR - used another branch. Have created a PR - https://github.com/xchem/fragalysis-backend/pull/185.

**Pull request closed without merging**